lib/download.js

```diff
@@ -2,15 +2,40 @@
 
 let fs      = require('fs');
 let https   = require('https');
+let url     = require('url');
+let cli     = require('heroku-cli-util');
+let tunnel  = require('tunnel-agent');
 
-function file (url, path, opts) {
+function file (urlStr, path, opts) {
   let tmpPath = path + '.tmp';
   return new Promise(function (fulfill, reject) {
-    https.get(url, function (res) {
+    let httpsProxy = process.env.HTTPS_PROXY || process.env.https_proxy;
+    let agent;
+    if (httpsProxy) {
+      cli.hush(`proxy set to ${httpsProxy}`);
+      let proxy = url.parse(httpsProxy);
+
+      agent = tunnel.httpsOverHttp({
+        proxy: {
+          host: proxy.hostname,
+          port: proxy.port || 8080
+        }
+      });
+    } else {
+      agent = new https.Agent();
+    }
+
```

Comment:
I'll bet you're already considering the refactor; for this, it would be great to just have:

```js
let agent = getAgent(process.env.HTTPS_PROXY || process.env.https_proxy);
```

Then a separate function like:

```js
function getAgent(proxy) {
  if (!proxy) return new https.Agent();
  cli.hush(`proxy set to ${proxy}`);
  ... etc ...
  return tunnel.httpsOverHttp({ ... });
}
```